cmd/src/batch_common.go

```diff
@@ -268,9 +268,14 @@ func executeBatchSpec(ctx context.Context, ui ui.ExecUI, opts executeBatchSpecOp
 		return err
 	}
 
-	// In the past, we checked `docker version`, but now we retrieve the number
-	// of CPUs, since we need that anyway and it performs the same check (is
-	// Docker working _at all_?).
+	// In the past, we relied on `getBatchParallelism` to ascertain if docker is running,
+	// however, we don't always check for the number of CPUs (especially when the -j parallelis)
+	// flag is passed. This is a more explicit check to confirm docker is working.
+	if err := checkExecutable("docker", "version"); err != nil {
+		fmt.Println("docker info thing")
+		return err
+	}
+
 	parallelism, err := getBatchParallelism(ctx, opts.flags.parallelism)
```

Comment:
[One of the reasons why this wasn't working for the docker check is that, when the `-j` flag is passed, we don't bother to check the number of CPUs on the user's machine, therefore failing to confirm if docker is running or not](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/src-cli/-/blob/cmd/src/batch_common.go?L579:1&popover=pinned).

Reply:
I particularly appreciate the updated comment. 😆